test/api/qcFlags.test.js

```diff
@@ -0,0 +1,218 @@
+
+    describe('GET /api/qcFlags', () => {
+        it('should successfuly fetch all data', async () => {
+            const response = await request(server).get('/api/qcFlags');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 5, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(5);
+            const oneFetchedFlag = data.find(({ id }) => id === 4);
+            expect(oneFetchedFlag).to.be.eql({
+                id: 4,
+                from: (1647914400 + 10000) * 1000,
+                to: (1647914400 + 10000) * 1000,
+                comment: 'Some qc comment 4',
+                createdAt: 1707825439000,
+                updatedAt: 1707825439000,
+                runNumber: 1,
+                dplDetectorId: 1,
+                createdById: 2,
+                createdBy: { id: 2, externalId: 456, name: 'Jan Jansen' },
+                flagTypeId: 13,
+                flagType: { id: 13, name: 'Bad', method: 'Bad', bad: true, archived: false, color: null },
+            });
+        });
+        it('should successfuly filter on ids', async () => {
+            const response = await request(server).get('/api/qcFlags?filter[ids][]=1');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 1, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(1);
+            expect(data[0].id).to.be.equal(1);
+        });
+        it('should successfuly filter on dataPassIds, runNumbers, dplDetectorIds', async () => {
+            const response = await request(server)
+                .get('/api/qcFlags?filter[dataPassIds][]=1&filter[runNumbers][]=106&filter[dplDetectorIds][]=1');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 3, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(3);
+            expect(data.map(({ id }) => id)).to.have.all.members([1, 2, 3]);
+        });
+
+        it('should successfuly filter on simulationPassIds, runNumbers, dplDetectorIds', async () => {
+            const response = await request(server)
+                .get('/api/qcFlags?filter[simulationPassIds][]=1&filter[runNumbers][]=106&filter[dplDetectorIds][]=1');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 1, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(1);
+            expect(data[0].id).to.be.equal(5);
+        });
+
+        it('should retrive no records when filtering on ids', async () => {
```

Comment:
filtering on non-existing ids

I would say that you can merge this with the other filtering on id, up to you

Reply:
renamed